x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -511,10 +514,54 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
       />
     );
   }, [kibanaDataViews]);
+
+  const tourConfig = {
+    currentTourStep: 1,
+    isTourActive: true,
+    tourPopoverWidth: 300,
+    tourSubtitle: i18n.DATA_SOURCE_GUIDE_SUB_TITLE,
+  };
+
+  const [tourState, setTourState] = useState(() => {
+    const tourStateString = localStorage.getItem(DATA_VIEW_TOUR_ACTIVE_KEY);
+
+    if (tourStateString != null) {
+      return JSON.parse(tourStateString);
+    }
+    return tourConfig;
+  });
+
+  useEffect(() => {
+    // Store the tour data
+    localStorage.setItem(DATA_VIEW_TOUR_ACTIVE_KEY, JSON.stringify(tourState));
+  }, [tourState]);
+
   const DataSource = useMemo(() => {
+    const demoTourSteps = [
+      {
+        step: 1,
+        title: i18n.DATA_SOURCE_GUIDE_TITLE,
+        content: (
+          <span>
+            <p>{i18n.DATA_SOURCE_GUIDE_CONTENT}</p>
+            <EuiSpacer />
+          </span>
+        ),
+      },
+    ];
+    const finishTour = () => {
```

Comment:
This might as well be `useCallback`'d if you're using it as a prop below.

Reply:
I think this is resolved here.. 992d397dcf49356d421dce1dbcb4fec84336b543